project/src/main/webapp/document.jsp

```diff
@@ -189,6 +196,18 @@
       function getHash() {
         return window.location.hash.substr(2);
       }
+
+      //Shares document with email from share-form
+      function share() {
```

Comment:
just curious; why not just create a normal POST request using action="/Servlet" method="POST"?

Reply:
This way, the page isn't reloaded when the form is submitted